ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -1884,7 +1905,11 @@ extension BrowserViewController: WKUIDelegate {
       )
 
       tab.upgradedHTTPSRequest = nil
-      braveCore.httpsUpgradeExceptionsService.addException(for: originalURL)
+      if let httpsUpgradeService = HttpsUpgradeServiceFactory.get(privateMode: tab.isPrivate),
+        let host = originalURL.host
+      {
+        httpsUpgradeService.allowHttp(forHost: host)
```

Comment:
Do we want to add a failed upgrade to the allowlist? It doesn't follow the same logic that [Chromium has](https://source.chromium.org/chromium/chromium/src/+/main:ios/chrome/browser/https_upgrades/model/https_only_mode_upgrade_tab_helper.mm;l=255;bpv=0;bpt=1) but maybe thats fine?

Reply:
@arthuredelstein had recommended an allow list for standard & strict mode for perf reasons if we are clearing the allow list when preference changes. I think this would only be an issue if the site fixes (or adds) HTTPS before the user's allowlist is reset (time based expiry, clear browser history, or pref change)?